src/rpc/blockchain.cpp

```diff
@@ -1545,34 +1546,141 @@ UniValue MempoolInfoToJSON(const CTxMemPool& pool)
     ret.pushKV("mempoolminfee", ValueFromAmount(std::max(pool.GetMinFee(maxmempool), ::minRelayTxFee).GetFeePerK()));
     ret.pushKV("minrelaytxfee", ValueFromAmount(::minRelayTxFee.GetFeePerK()));
     ret.pushKV("unbroadcastcount", uint64_t{pool.GetUnbroadcastTxs().size()});
+
+    if (feeLimits) {
+        const MempoolHistogramFeeRates& limits{feeLimits.value()};
+
+        /* Keep histogram per...
+         * ... cumulated tx sizes
+         * ... txns (count)
+         * ... cumulated fees */
+        std::vector<uint64_t> sizes(limits.size(), 0);
+        std::vector<uint64_t> count(limits.size(), 0);
+        std::vector<CAmount> fees(limits.size(), 0);
+
+        for (const CTxMemPoolEntry& e : pool.mapTx) {
+            const CAmount fee{e.GetFee()}, afees{e.GetModFeesWithAncestors()}, dfees{e.GetModFeesWithDescendants()};
+            const size_t size{e.GetTxSize()}, asize{e.GetSizeWithAncestors()}, dsize{e.GetSizeWithDescendants()};
```

Comment:
tx size is limited to uint32_t on all platforms on master

```suggestion
            const uint32_t size{e.GetTxSize()}, asize{e.GetSizeWithAncestors()}, dsize{e.GetSizeWithDescendants()};
```

Reply:
Thank you, I have tried to address that to avoid subsequent narrowing warnings.

Reply:
Putting such an assumption here is unnecessarily bug-prone for no benefit. All the functions involved return 64-bit types, so should not be truncated. Furthermore, GetSizeWith* may very well include multiple transactions. `size_t` is the right type here IMO.

Reply:
> Putting such an assumption here is unnecessarily bug-prone for no benefit.

Then #21848 should be reverted first. Otherwise using size_t here is compiled down to the same code anyway.